packages/extension-link/src/link.ts

```diff
@@ -351,7 +351,12 @@ export const Link = Mark.create<LinkOptions>({
 
           if (text) {
             const { protocols, defaultProtocol } = this.options
-            const links = find(text).filter(
+            // Prosemirror replaces zero-width non-joiner characters
+            // with Object Replacement Character from unicode.
+            // Therefore, linkifyjs does not recognize the
+            // link properly. We replace these characters
+            // with regular spaces to fix this issue.
+            const links = find(text.replaceAll('\uFFFC', ' ')).filter(
```

Comment:
Did we figure out if this char is valid in a URL? If not, may want to try to upstream this to linkify

Reply:
Reported to `linkifyjs` https://github.com/nfrasser/linkifyjs/issues/495